scripts/s3/migrate_to_external_accounts.py

```diff
@@ -0,0 +1,345 @@
+import logging
+import os
+import sys
+import urlparse
+
+from nose.tools import *  # noqa
+from modularodm import Q
+
+from framework.mongo import database
+from framework.transactions.context import TokuTransaction
+
+from website import settings
+from website.app import init_app
+from website.models import User, Node
+from website.oauth.models import ExternalAccount
+from website.addons.s3 import settings as s3_settings
+from website.addons.s3 import utils
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+PROVIDER = 's3'
+PROVIDER_NAME = 'Amazon S3'
+ENCRYPT_UPLOADS = s3_settings.ENCRYPT_UPLOADS_DEFAULT
+
+# set of {ExternalAccount._id: [old_user_settings._id, ...]} mappings
+# with invalid credentials, for logging purposes
+invalid_oauth_creds = {}
+
+
+def verify_user_settings_documents(user_document):
+    try:
+        assert_in('_id', user_document)
```

Comment:
I don't think we install nose in production environments. You can just use plain `assert` statements instead.